internal/config/config.go

```diff
@@ -1,34 +1,72 @@
 package config
 
 import (
-	"flag"
 	"fmt"
+	"log"
+	"nubedb/pkg/resolver"
+	"os"
+	"strings"
 )
 
-type Config struct {
+type NodeCfg struct {
 	ID               string
-	Host             string
 	ApiPort          int
 	ApiAddress       string
 	ConsensusPort    int
 	ConsensusAddress string
+	GrpcPort         int
+	GrpcAddress      string
+}
+
+type Config struct {
+	CurrentNode NodeCfg
+	Nodes       map[string]NodeCfg
 }
 
 func New() Config {
-	id := flag.String("id", "node_1", "")
-	host := flag.String("host", "localhost", "")
-	apiPort := flag.Int("api-port", 3001, "")
-	consensusPort := flag.Int("consensus-port", 4001, "")
-	flag.Parse()
-
-	apiAddress := fmt.Sprintf("%s:%v", *host, *apiPort)
-	consensusAddress := fmt.Sprintf("%s:%v", *host, *consensusPort)
+	currentNodeID := os.Getenv("NODE")
```

Comment:
Should the new function crash the application or return errors?

Reply:
Now it does: https://github.com/narvikd/nubedb/pull/10/commits/8657625d908d5476faa4ce3092499a6b01db50ee